src/alert.cpp

```diff
@@ -1829,5 +1829,20 @@ namespace libtorrent {
 		}
 	}
 
+	dht_direct_response_alert::dht_direct_response_alert(
+		aux::stack_allocator&, void* userdata
+		, udp::endpoint const& addr, entry const& response)
```

Comment:
here's a crazy idea: what if the alert had the raw bencoded buffer (allocated with aux::stack_allocator) instead of an entry object here? It would be a lot cheaper to manage the memory, and decoding of the buffer could be done lazily.